testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,693 @@
+        if (currentTestStates.size() == MAX_NUM) {
+            if (enableLogging) {
+                logger.startTimer();
+            }
+            if (enableLogging) {
+                logger.stopTimeAndUpdate(StateCaptureLogger.Task.DIFF);
+                logger.saveToFileAndReset();
+            }
+        }
+    }
+
+    Set<String> File2SetString(String path) {
+        File file = new File(path);
+        Set<String> keys = new HashSet<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                keys.add(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return keys;
+    }
+
+    public void setup() {
+        subxmlFold = Configuration.config().getProperty("replay.subxmlFold");
+        rootFold = Configuration.config().getProperty("replay.rootFold");
+        diffFold = Configuration.config().getProperty("replay.diffFold");
+        slug = Configuration.config().getProperty("replay.slug");
+        reflectionFile = Configuration.config().getProperty("replay.reflectionFold") + "/reflection.txt";
+    }
+
+    public void fixing(String fieldName) throws IOException {
+        String subxml0 = subxmlFold + "/passing_order_xml";
+        try {
+            String path0 = subxml0 + "/" + fieldName + ".xml";
+            String state0 = readFile(path0);
+            String className = fieldName.substring(0, fieldName.lastIndexOf("."));
+            String subFieldName = fieldName.substring(fieldName.lastIndexOf(".")+1, fieldName.length());
+
+            Object ob_0;
+            XStream xstream = getXStreamInstance();
+
+            try {
+                Class c = Class.forName(className);
+                Field[] fieldList = c.getDeclaredFields();
+                for (int i=0; i< fieldList.length; i++) {
+                    if (!fieldList[i].getName().equals(subFieldName)) {
+                        continue;
+                    }
+                    try {
+                        fieldList[i].setAccessible(true);
+                        Field modifiersField = Field.class.getDeclaredField("modifiers");
+                        modifiersField.setAccessible(true);
+                        modifiersField.setInt(fieldList[i], fieldList[i].getModifiers() & ~Modifier.FINAL);
+                    }
+                    catch (Exception e) {
+                        String outputPrivateError = fieldName + " reflectionError: " + e + "\n";
+                        Files.write(Paths.get(reflectionFile), outputPrivateError.getBytes(),
+                                StandardOpenOption.APPEND);
+                    }
+                    try {
+                        ob_0 = fieldList[i].get(null);
+                        boolean threadLocal = false;
+                        if (ob_0 instanceof ThreadLocal) {
```

Comment:
Would be good to have a comment here to remind ourselves that this logic with `ThreadLocal` is just so we can serialize/deserialize the object wrapped within a `ThreadLocal`.